### ISSUE TITLE
a11y Review List

### ISSUE THREAD

**Issue by mathetos** [WordImpress/Give#325]
I'm making this into a list of a11y issues that were brought up in an accessibility review by Jordan Quintal. Each item will be linked to it's respective issue and marked off as they are completed. That way we know the entire first review is complete.

- [x] `<input id="give-amount">` creates multiple IDs on the page when multiple forms are present (Issue #326 )
- [x] `<input class="give-text-input" name="give-amount">` doesn't have a label (Issue #328)
- [x] `id="give-form-{ID}"`is output twice on the page, once as a div, once as the form ID (Issue #329)
- [x] `id="give-amount"` is output twice on the page (Issue #330)
- [x] Multi-level select field has no label (Issue #331)
- [x] Login form has a `<p>` with no text in it. It's the `<p>` for the Submit Button (Issue #332)
- [x] Make ALL ids unique to avoid identical IDs when multiple forms are on the same page. (Issue #379)

This is the whole list boiled down. I'll add links to each item as I create the Issues for them.

**Comment by DevinWalker** [comment]
Nice, I didn't know you create a master task like this with items attached :+1:

**Comment by michaelbeil** [comment]
fancy shmancy. kudos on the a11y improvements!

**Comment by mathetos** [comment]
Thanks @michaelbeil ! Overall, we're pretty happy with the fact that it seems like only minor improvements needed.

**Comment by mathetos** [comment]
@DevinWalker All Issues now currently have PR's. Please review and merge. Then we can upload the new master to the demo site and @TheGeniusWM can do a revised #a11y review to see if we've resolved all the things. Thanks!

**Comment by mathetos** [comment]
Just added Issue #379 to the list which already has a PR but it's pretty significant so it will need careful inspection. Heads up on that @DevinWalker

**Comment by DevinWalker** [comment]
@mathetos can you iron out the conflicts in your PR and resubmit when you get the chance?

**Comment by ramiy** [comment]
#858

**Comment by ramiy** [comment]
@mathetos,

Please review core `title` attributes in `a` tags. We should either remove the `title` attributes, or replace them with `aria-label` attributes.

https://core.trac.wordpress.org/changeset/25743
https://core.trac.wordpress.org/changeset/29583
https://core.trac.wordpress.org/changeset/32981

**Comment by ramiy** [comment]
One more fix we need to implement is the "close button". Remove the title attribute and move the text into a span with `class='screen-reader-text'` like it's done in: https://core.trac.wordpress.org/changeset/37531

**Comment by ramiy** [comment]
@mathetos 

Check out this commit - https://core.trac.wordpress.org/changeset/37680

When writing strings, help text shouldn't assume users can see.

We should avoid reference to "visual" positions in favor of positions in the document structure.

**Comment by mathetos** [comment]
Thanks to the initial #a11y review from @TheGeniusWM and then the very detailed Admin and front-end follow-up from @ramiy Give is now very thoroughly #a11y-compliant. The latest changes in version 1.7 will complete this effort. It'll be up to us to continue to maintain this status as the plugin continues to develop.

Thanks so much guys!
